### PR TITLE
Fixed isolated nodes handling in node2vec

### PR DESCRIPTION
**Pull request by dokato**
- [x] Does this PR have a descriptive title that could go in our release notes?
- [x] Does this PR add any new dependencies?
- [x] Does this PR modify any existing APIs?
   - [x] Is the change to the API backwards compatible?
- [x] Have you built the documentation (reference and/or tutorial) and verified the generated documentation is appropriate?

#### Reference Issues/PRs

Fixes issue #952 

#### What does this implement/fix? Briefly explain your changes.

It removes isolated nodes from `labels` and warns users that such cases were detected.

### CODE REVIEW

**Comment by bdpedigo** [comment]
not sure why the docs didnt build here, going to rerun and just see if it was a fluke

**Comment by bdpedigo** [comment]
believe these upstream changes should fix the bug in doc building unrelated to this PR

**Review by bdpedigo — APPROVED** [review]

File: graspologic/embed/n2v.py [hunk 1]
```diff
@@ -142,6 +143,10 @@ def node2vec_embed(
 
     labels = list(node2vec_graph.original_graph.nodes())
     remapped_labels = node2vec_graph.label_map_to_string
+    isolated_nodes = [x for x in nx.isolates(node2vec_graph.original_graph)]
+    if len(isolated_nodes):
```
Comment (daxpryce) [inline graspologic/embed/n2v.py:147]: my preference is for explicit booleanification not implicit truthiness - so `len(isolated_nodes) > 0`

File: graspologic/embed/n2v.py [hunk 2]
```diff
@@ -142,6 +143,10 @@ def node2vec_embed(
...
+        warnings.warn(f"Isolated nodes found: {isolated_nodes}")
+        labels = list(np.setdiff1d(labels, isolated_nodes))
```
Comment (daxpryce) [inline graspologic/embed/n2v.py:149]: neat - TIL about setdiff1d

**Review by daxpryce — CHANGES_REQUESTED** [review]
Minor change then lgtm!

**Comment by bdpedigo** [comment]
@daxpryce how do you feel about a ✅ ?

**Review by daxpryce — APPROVED** [review]

**Comment by daxpryce** [comment]
> @daxpryce how do you feel about a ✅ ?

https://www.youtube.com/watch?v=lFeLDc2CzOs

super green.

**Comment by bdpedigo** [comment]
thanks @dokato!

**Pull request merged**